api/swagger.yml

```diff
@@ -2485,90 +2573,94 @@ paths:
         - in: path
           name: repository
           required: true
-          type: string
+          schema:
+            type: string
         - in: path
           name: run_id
           required: true
-          type: string
+          schema:
+            type: string
         - in: path
           name: hook_run_id
           required: true
-          type: string
-      produces:
-        - application/octet-stream
+          schema:
+            type: string
       responses:
         200:
           description: run hook output
-          schema:
-            type: file
+          content:
+            application/octet-stream:
+              schema:
+                type: string
+                format: binary
         401:
-          $ref: "#/responses/Unauthorized"
+          $ref: "#/components/responses/Unauthorized"
         404:
-          description: output not found
-          schema:
-            $ref: "#/definitions/error"
+          $ref: "#/components/responses/NotFound"
         default:
-          description: generic error response
-          schema:
-            $ref: "#/definitions/error"
+          $ref: "#/components/responses/ServerError"
 
   /repositories/{repository}/metadata/meta_range/{meta_range}:
     parameters:
       - in: path
         name: repository
         required: true
-        type: string
+        schema:
+          type: string
       - in: path
         name: meta_range
         required: true
-        type: string
+        schema:
+          type: string
     get:
       tags:
         - metadata
       operationId: getMetaRange
       summary: return URI to a meta-range file
-      # TODO(ariels): Actually redirect to S3.
       responses:
         200:
           description: meta-range URI
-          headers:
-            Location:
-              type: string
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/StorageURI'
         401:
-          $ref: "#/responses/Unauthorized"
+          $ref: "#/components/responses/Unauthorized"
+        404:
+          $ref: "#/components/responses/NotFound"
         default:
-          description: generic error response
-          schema:
-            $ref: "#/definitions/error"
+          $ref: "#/components/responses/ServerError"
 
   /repositories/{repository}/metadata/range/{range}:
     parameters:
       - in: path
         name: repository
         required: true
-        type: string
+        schema:
+          type: string
       - in: path
         name: range
         required: true
-        type: string
+        schema:
+          type: string
     get:
       tags:
         - metadata
       operationId: getRange
       summary: return URI to a range file
-      # TODO(ariels): Actually redirect to S3.
       responses:
         200:
           description: range URI
-          headers:
-            Location:
```

Comment:
This moves us further away from REST.  Ideally this could be a redirect, making it more HTTPish and also making it easier to implement some clients -- e.g. if/once we decide to use presigned URLs.

Reply:
missed this one - thanks